### PR TITLE
I don't like qt creator no more

### PR DESCRIPTION
**Pull request by MCLiii**

### CODE REVIEW

**Comment by jackwherry** [comment]
Are we replacing qmake with cmake? I'm sure you know this already but you can use qmake without Qt Creator (that's what I'm doing on my laptop).

**Review by jrvollmer — CHANGES_REQUESTED** [review]
This is not a good use of time. Just use qt creator or your terminal in vscode

**Review by MCLiii — COMMENTED** [review]
I think there's a lot to be excited about this change

- first by changing from qmake to CMake we can allow a hastle free setup and have autocomplete working on many ides 

- second cmake would mean less bloat for our software system, imagine having to download 20gigs of qt dependencies, yuck! Now we can just tell members brew install this apt install that!
- third by using cmake we will be able to save 2 seconds of compile time incomparison to using qmake, this will be equal to a whole hour after 1800 compile! massively improving our work flow!

**Review by jrvollmer — CHANGES_REQUESTED** [review]
Okay, I don't actually have any qualms with using cmake instead of qmake. However, I think we should use one or the other to avoid any hassle/issues with supporting both. **@MCLiii, if you want to transition to cmake, please take care of the following:**
- [x] Remove the `solar-car-dashboard.pro[.user]` files.
- [x] Make sure to test building and running the latest version of the dashboard on the pi using these changes. Not that I think it's going to pose any serious issues. I just want to make sure the pi is good to go before pushing the changes through.
- [x] Update the necessary dependencies and run instructions in the README.

Also, to add to your points @MCLiii, since cmake is more generic than qmake, it could be a better learning experience for members.

**Comment by jackwherry** [comment]
> second cmake would mean less bloat for our software system, imagine having to download 20gigs of qt dependencies, yuck! Now we can just tell members brew install this apt install that!

Contributors are still going to need the Qt .so/.dylib/.dll and header files. `brew install qt` was already enough to get all that and qmake installed and it definitely wasn’t measured in gigabytes (it's 665 MiB on my machine no matter what build system you're using) 
image

I'm also not opposed in principle to switching (James is right that it's probably a better learning experience because cmake is more widely used anyway).

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
+set(CMAKE_AUTOUIC ON)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+
+cmake_minimum_required(VERSION 3.5)
+project(solar-car-dashboard)
+
+# Find the required Qt packages
+find_package(QT NAMES Qt5 Qt6 COMPONENTS Core Quick)
```
Comment (jrvollmer) [inline CMakeLists.txt:9]: Why are we adding qt5 here? Just wondering if something was added/changed to make it necessary

File: README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@
 1. Once you have an SSH key, clone the [sc1-data-format repository](https://github.com/badgerloop-software/sc1-data-format) to your computer. Make sure to clone it using SSH (when you go to copy the clone link, there will be an SSH option above the link).
 2. Next, `cd` into the `sc1-driver-io` repository and run `git submodule update --init`.
 
-### Running the Dashboard
+### Running the Dashboard with QMake
```
Comment (jrvollmer) [inline README.md]: This was part of the checklist of things. I think we should stick with one build system. If you want to move forward with cmake, let's remove qmake (this section of the readme and solar-car-dashboard.pro) so we don't have to maintain both.

**Review by jrvollmer — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (MCLiii) [reply CMakeLists.txt:9]: The linux apt package of qt6 does not currently have quick module I believe, thus it's looking for both versions

Comment (jrvollmer) [reply CMakeLists.txt:9]: Sg

File: backendprocesses.h [hunk 3]
```diff
@@ -4,6 +4,10 @@
 #undef unix
 #endif
 
+#ifdef unix
```
Comment (jrvollmer) [inline backendprocesses.h]: Sorry to keep this PR going, but let's remove this. It's already done a few lines above

**Review by jrvollmer — APPROVED** [review]
Beautiful. I love it. Thanks Mingcan, very cool!

**Pull request merged**